src/spag/main.rs

```diff
@@ -22,21 +27,36 @@ use super::yaml_util;
 
 
 pub fn main() {
-    let args: Args = args::parse_args();
-    if args.cmd_request {
-        spag_request(&args);
-    } else if args.cmd_history {
-        spag_history(&args);
-    } else if args.cmd_env {
-        spag_env(&args);
-    } else if args.cmd_get || args.cmd_post || args.cmd_put || args.cmd_patch || args.cmd_delete {
-        spag_method(&args);
+    let argv: Vec<String> = std::env::args().collect();
+    let main_argv: Vec<String> = std::env::args().take(2).collect();
+    let args: MainArgs = args::parse_main_args(&main_argv);
+    match args.arg_command.as_str() {
+        "env" => {
+            spag_env(&args::parse_env_args(&argv))
+        },
+        "r" | "request" => {
+            spag_request(&args::parse_request_args(&argv))
+        },
+        "history" => {
+            spag_history(&args::parse_history_args(&argv))
+        },
+        "get" | "post" | "put" | "patch" | "delete" => {
+            spag_method(&args::parse_method_args(&argv))
+        },
+        command if command.is_empty() => {
+            printerrln!("Received no command or options");
```

Comment:
It would be cool if you could do this when there was no command for the `env`, `request`, `history` ones too. Instead of having it say `Invalid Arguments` and not print the whole help string for the command. 

It's cool to print the help for a command when you do `spag env` for example. Can't do that with `spag history` because that actually does at thing, but still.